### ISSUE TITLE
App immediately closes without --no-sandbox and launches to a black window on reinstall

### ISSUE THREAD

**Issue by ruthslambat** [itchio/itch#2816]
The app launches and immediately closes without `--no-sandbox`. The first time it did this, I assumed the installer had gone wrong, so I reinstalled. After this, the app launched and went to a black screen. After spending about an hour messing with config files and whether or not I specified `--no-sandbox`, I deleted all itch-related folders and reinstalled once more. It finally worked with this truly fresh install when I specified `--no-sandbox`.

Output (outputs for all previous attempts, whether immediate closes or black windows, were identical save for the presence or lack thereof of `"--no-sandbox"`:

`2022/05/01 20:58:20 itch-setup will log to /tmp/itch-setup-log.txt`
`2022/05/01 20:58:20 =========================================
2022/05/01 20:58:20 itch-setup "v1.26.0, built on Apr 21 2021 @ 21:48:12, ref 48f97b3e7b0b065a2478811b8d0ebcae414845fd" starting up at "2022-05-01 20:58:20.100234464 -0400 EDT m=+0.001866835" with arguments:`
`2022/05/01 20:58:20 "/home/ruthslambat/.itch/itch-setup"`
`2022/05/01 20:58:20 "--prefer-launch"`
`2022/05/01 20:58:20 "--appname"`
`2022/05/01 20:58:20 "itch"`
`2022/05/01 20:58:20 "--"`
`2022/05/01 20:58:20 =========================================`
`2022/05/01 20:58:20 App name specified on command-line: itch`
`2022/05/01 20:58:20 Locale:  en-US`
`2022/05/01 20:58:20 Initializing installer GUI...`
`2022/05/01 20:58:20 Using GTK UI`
`2022/05/01 20:58:20 Initializing (itch) multiverse @ (/home/ruthslambat/.itch)`
`2022/05/01 20:58:20 (/home/ruthslambat/.itch)(current = "25.5.1", ready = "")`
`2022/05/01 20:58:20 Launch preferred, attempting...`
`2022/05/01 20:58:20 Launching (25.5.1) from (/home/ruthslambat/.itch/app-25.5.1)`
`2022/05/01 20:58:20 Kernel should support SUID sandboxing, leaving it enabled`
`2022/05/01 20:58:20 App launched, getting out of the way`

OS: openSUSE Tumbleweed 20220428

**Comment by proteanblank** [comment]
Same problem.  App launches and immediately closes without --no-sandbox.  
different OS though, PRETTY_NAME="Pop!_OS 22.04 LTS"  (an Ubuntu/Debian derived flavor)
seems to be SUID related here as well (guess, not investigated on my system to any depth yet)

**Comment by anidotnet** [comment]
Same problem here with Ubuntu 22.04 LTS.. If I directly run the app from "/home/anindya/.itch/app-25.5.1" directory, I get below error log of electron:

```
% ./itch                                                                                                                                                                                                           
09:17:30.344 INFO itch@25.5.1 on electron@10.4.5 in production
09:17:30.346 DEBUG (m/reac/preb/load-preferences) imported preferences: {"downloadSelfUpdates":true,"offlineMode":false,"installLocations":{},"defaultInstallLocation":"appdata","isolateApps":false,"closeToTray":true,"readyNotification":true,"showAdvanced":false,"openAtLogin":false,"openAsHidden":false,"manualGameUpdates":false,"preventDisplaySleep":true,"preferOptimizedPatches":false,"disableBrowser":false,"enableTabs":false}
09:17:30.471 DEBUG (m/reac/preb/load-preferences) imported preferences: {"downloadSelfUpdates":true,"offlineMode":false,"installLocations":{},"defaultInstallLocation":"appdata","isolateApps":false,"closeToTray":true,"readyNotification":true,"showAdvanced":false,"openAtLogin":false,"openAsHidden":false,"manualGameUpdates":false,"preventDisplaySleep":true,"preferOptimizedPatches":false,"disableBrowser":false,"enableTabs":false}
09:17:30.476 DEBUG (m/net/register-itch-protocol) Registering itch: protocol for session with user agent Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) itch/25.5.1 Chrome/85.0.4183.121 Electron/10.4.5 Safari/537.36
09:17:30.512 INFO (m/reac/preboot) preboot ran in 40.00ms
09:17:30.513 DEBUG (m/reac/open-at-login) Updating login item settings, open=false, hidden=false
09:17:30.513 DEBUG (m/reac/open-at-login) Copying /usr/share/applications/io.itch.itch.desktop => /home/anindya/.config/autostart/io.itch.itch.desktop
09:17:30.519 DEBUG (m/reac/winds) Waiting on root window to dispatch boot
09:17:31.480 INFO (m/reac/i18n) Language settings: preference undefined, sniffed en-GB, fallback en
09:17:31.483 DEBUG (m/reac/locales) Reading local locale file /home/anindya/.itch/app-25.5.1/resources/app/src/static/locales/en.json
09:17:31.486 DEBUG (m/reac/locales) Reading remote locale file /home/anindya/.itch/app-25.5.1/resources/app/src/static/locales/en.json
[39746:0511/144731.852380:FATAL:gpu_data_manager_impl_private.cc(442)] GPU process isn't usable. Goodbye.
[1]    39746 trace trap (core dumped)  ./itch

```

and this is my system details

```
Operating System: Ubuntu 22.04 LTS                
Kernel: Linux 5.15.0-27-generic
Architecture: x86-64

```

Hope this helps to fix the issue. Meantime, I have found a workaround, the below command runs the executable just fine.

```
% ./itch --in-progress-gpu --no-sandbox  
```

**Comment by Jeffrey04** [comment]
this is likely a related to https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1944468

**Comment by OtherCannon** [comment]
Also experiencing this issue on Pop OS 22.04, using --no-sandbox is the only way to get the app to run

**Comment by CobaltSpace** [comment]
Unchecking "Disable hardware accelerations" in Preferences fixed this for me.  I used `--in-process-gpu` to get in and access preferences, and now I don't need it.

**Comment by tobil4sk** [comment]
I think this is a duplicate of #2771?

**Comment by leafo** [comment]
These issues should be resolved with the most recent of the app from an Electron update. Please leave a comment if you are still having issues with 26.9.0 or above.
